pkg/controller/storageversionmigrator/storageversionmigrator.go

```diff
@@ -251,31 +261,56 @@ func (svmc *SVMController) sync(ctx context.Context, key string) error {
 		return fmt.Errorf("GC cache is not up to date, requeuing to attempt again. gcListResourceVersion: %s, listResourceVersion: %s", gcListResourceVersion, listResourceVersion)
 	}
 
-	toBeProcessedSVM, err = svmc.kubeClient.StoragemigrationV1alpha1().
+	toBeProcessedSVM, err = svmc.kubeClient.StoragemigrationV1beta1().
 		StorageVersionMigrations().
 		UpdateStatus(
 			ctx,
-			setStatusConditions(toBeProcessedSVM, svmv1alpha1.MigrationRunning, migrationRunningStatusReason, ""),
+			setStatusConditions(toBeProcessedSVM, svmv1beta1.MigrationRunning, migrationRunningStatusReason, ""),
 			metav1.UpdateOptions{},
 		)
 	if err != nil {
 		return err
 	}
 
-	gvk, err := svmc.restMapper.KindFor(gvr)
+	// ToDo: implement a mechanism to resume migration from the last migrated resource in case of a failure
+	// process storage migration
+	err, failedMigration := svmc.runMigration(ctx, logger, *gvr, resourceMonitor, toBeProcessedSVM, listResourceVersion)
+	if err != nil {
+		return err
+	}
+	if failedMigration {
+		return nil
+	}
+
+	_, err = svmc.kubeClient.StoragemigrationV1beta1().
+		StorageVersionMigrations().
+		UpdateStatus(
+			ctx,
+			setStatusConditions(toBeProcessedSVM, svmv1beta1.MigrationSucceeded, migrationSuccessStatusReason, ""),
+			metav1.UpdateOptions{},
+		)
 	if err != nil {
 		return err
 	}
 
+	logger.V(4).Info("Finished syncing svm resource", "key", key, "grf", gr.String(), "elapsed", time.Since(startTime))
+	return nil
+}
+
+func (svmc *SVMController) runMigration(ctx context.Context, logger klog.Logger, gvr schema.GroupVersionResource, resourceMonitor *garbagecollector.Monitor, toBeProcessedSVM *svmv1beta1.StorageVersionMigration, listResourceVersion string) (err error, failed bool) {
 	for _, obj := range resourceMonitor.Store.List() {
 		accessor, err := meta.Accessor(obj)
 		if err != nil {
-			return err
+			return err, false
+		}
+		gvk, err := svmc.restMapper.KindFor(gvr)
```

Comment:
Do this once outside of the loop.

Reply:
Yep, done. Also should this one also fail the migration? I'll set it as failed for now but wdyt?